x-pack/legacy/plugins/infra/public/utils/url_state.tsx

```diff
@@ -81,6 +82,14 @@ class UrlStateContainerLifecycle<UrlState> extends React.Component<
     const newUrlState = mapToUrlState(decodeRisonUrlState(newUrlStateString));
 
     onInitialize(newUrlState);
+
+    // When the newURLState is empty we can assume that the state will becoming
+    // from the urlState initially. By setting populateWithIntialState to true
+    // this will now serialize the initial urlState into the URL when the page is
+    // loaded.
+    if (!newUrlState && this.props.populateWithInitialState) {
```

Comment:
This seems to work well :+1: The only downside I can see is that `onInitialize` is called with the "incorrect" empty state. On the metrics explorer that doesn't have any negative effect, but it could be problematic in other cases where it would cause an "empty" init followed by an update. How about making sure it's called with the `urlState` as in

```typescript
    // When the newURLState is empty we can assume that the state will be coming
    // from the urlState initially. By setting populateWithIntialState to true
    // this will now serialize the initial urlState into the URL when the page is
    // loaded.
    if (!newUrlState && this.props.populateWithInitialState) {
      this.replaceStateInLocation(urlState);
      onInitialize(urlState);
    } else {
      onInitialize(newUrlState);
    }
```